includes/class-wc-payments-webhook-reliability-service.php

```diff
@@ -60,9 +66,16 @@ public function __construct(
 		$this->action_scheduler_service   = $action_scheduler_service;
 		$this->webhook_processing_service = $webhook_processing_service;
 
-		add_action( 'woocommerce_payments_account_refreshed', [ $this, 'maybe_schedule_fetch_events' ] );
-		add_action( self::WEBHOOK_FETCH_EVENTS_ACTION, [ $this, 'fetch_events_and_schedule_processing_jobs' ] );
+		add_action( 'woocommerce_payments_account_refreshed', [ $this, 'maybe_schedule_failed_webhook_events' ] );
+		add_action( self::WEBHOOK_FETCH_EVENTS_ACTION, [ $this, 'schedule_processing_failed_webhook_events' ] );
 		add_action( self::WEBHOOK_PROCESS_EVENT_ACTION, [ $this, 'process_event' ] );
+
+		// Hook the scheduled action to the method.
+		add_action( self::WEBHOOK_PROCESS_EVENTS_ACTION, [ $this, 'load_and_process_events' ] );
```

Comment:
Hooking the event this way is dangerous. All parameters of the hook would be passed to `get_events`, leading to potential side effects.